goss.yaml

```
@@ -100,34 +104,62 @@ process:
     running: true
 
 command:
-  "aws --version":
+  aws --version:
     exit-status: 0
 
-  "git --version":
+  git --version:
     exit-status: 0
 
-  "git-lfs --version":
+  git-lfs --version:
     exit-status: 0
 
-  "/etc/cron.hourly/docker-low-disk-gc":
+  systemctl is-enabled docker-gc.timer:
     exit-status: 0
 
-  "/etc/cron.hourly/docker-gc":
+  /usr/local/bin/docker-gc:
     exit-status: 0
 
-  # Checks that docker is running
-  "docker info":
+  systemctl is-enabled docker-low-disk-gc.timer:
     exit-status: 0
-    timeout: 30000 # it can take some time for the daemon to start
 
-  # Checks that docker containers can run
-  "docker run --rm -v /var/run/docker.sock:/var/run/docker.sock docker:latest version":
+  /usr/local/bin/docker-low-disk-gc:
     exit-status: 0
-    timeout: 30000 # it can take some time to download the image
 
-  # Checks that permissions
-  'sh -c "docker run --rm -v \"$PWD:/pwd\" alpine:latest touch /pwd/test && stat -c %U:%G test"':
+  # Check docker userns is enabled
+  docker info --format='{{ `{{range .SecurityOptions}}{{if eq . "name=userns"}}{{"success"}}{{end}}{{end}}` }}':
     exit-status: 0
-    timeout: 30000 # it can take some time to download the image
+    timeout: 30000
+    stdout:
+    - success
+
+  # Check docker plugins are installed
+  docker info --format=$'{{ `{{range .ClientInfo.Plugins}}{{if eq .Name "buildx"}}success\n{{end}}{{if eq .Name "compose" }}success{{end}}{{end}}` }}':
+    exit-status: 0
+    timeout: 30000
+    stdout:
+    - success
+    - success
```

```suggestion
  # Check docker plugins are installed
  docker info --format=$'{{range .ClientInfo.Plugins}}{{if or (eq .Name "buildx") (eq .Name "compose")}}{{ .Name }},{{end}}{{end}}':
    exit-status: 0
    timeout: 30000
    stdout:
    - buildx,compose,
```

I don't know if this is any better, but i found the original to be a bit of a head-scratcher.  Also, this will tell you _which_ plugin is missing, if any.

WDYT?

---

Yeah, that's a great idea. I'll do that.

---

Hmm, I haven't actually been able to get goss to show the failing output, so you won't be able to tell which plugin is missing. There seems to be a fix in the works for goss v4 though: https://github.com/goss-org/goss/pull/576/files

But I found out that the elements of `stdout` are regexes and not expected lines. I think I can rewrite it with this in mind.